src/app/modules/shared_models/message_model.nim

```diff
@@ -248,45 +248,30 @@ QtObject:
       elif clock == self.items[i].clock: # break ties by message id
         if id > self.items[i].id:
           return i
-    return 0
+    return self.items.len
 
   proc filterExistingItems(self: Model, items: seq[Item]): seq[Item] =
     for item in items:
       if(self.findIndexForMessageId(item.id) < 0):
         result &= item
 
+  proc insertItemBasedOnClock*(self: Model, item: Item)
```

Comment:
Can you put the forward declaration at the top of the class, that's what we do usually.